File: CSharp/Concentus/Silk/Structs/TOCStruct.cs
using Concentus.Common.CPlusPlus;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Concentus.Silk.Structs
{
    /// <summary>
    /// Struct for TOC (Table of Contents)
    /// </summary>
    internal class TOCStruct
    {
        /// <summary>
        /// Voice activity for packet
        /// </summary>
        internal int VADFlag = 0;

        /// <summary>
        /// Voice activity for each frame in packet
        /// </summary>
        internal readonly Pointer<int> VADFlags = Pointer.Malloc<int>(SilkConstants.SILK_MAX_FRAMES_PER_PACKET);

        /// <summary>
        /// Flag indicating if packet contains in-band FEC
        /// </summary>
        internal int inbandFECFlag = 0;
    
        internal void Reset()
        {
            VADFlag = 0;
            VADFlags.MemSet(0, SilkConstants.SILK_MAX_FRAMES_PER_PACKET);
            inbandFECFlag = 0;
        }
    }
}

File: Go/opus/TOCStruct.go
package opus

type TOCStruct struct {
	VADFlag       int
	VADFlags      []int
	inbandFECFlag int
}

func NewTOCStruct() *TOCStruct {
	obj := &TOCStruct{}
	obj.VADFlags = make([]int, SilkConstants.SILK_MAX_FRAMES_PER_PACKET)
	return obj
}

func (t *TOCStruct) Reset() {
	t.VADFlag = 0
	MemSetLen(t.VADFlags, 0, SilkConstants.SILK_MAX_FRAMES_PER_PACKET)
	t.inbandFECFlag = 0
}
